packages/flutter/lib/src/cupertino/form_section.dart

```diff
@@ -0,0 +1,219 @@
+  ///
+  /// Defaults to the standard iOS padding, which is determined
+  /// using https://github.com/flutter/platform_tests/tree/master/ios_widget_catalog_compare.
+  final EdgeInsetsGeometry titlePadding;
+
+  /// Padding for the the [Column] encapsulating the [children].
+  ///
+  /// Defaults to zero padding if constructed with standard
+  /// [CupertinoFormSection] constructor. Defaults to the standard notched-style
+  /// iOS padding when constructing with [CupertinoFormSection.centered].
+  /// Determined using https://github.com/flutter/platform_tests/tree/master/ios_widget_catalog_compare.
+  final EdgeInsetsGeometry rowsPadding;
+
+  /// The list of rows in the section.
+  final List<Widget> children;
+
+  /// Sets the decoration for the section.
+  ///
+  /// Defaults to a [CupertinoColors.secondarySystemBackground] background
+  /// color.
+  final BoxDecoration decoration;
+
+  /// Sets the circular border radius for the [Column] encapsulating [children]
+  /// rows.
+  ///
+  /// Defaults to 0.0 when constructed with standard [CupertinoFormSection]
+  /// constructor. Defaults to 10.0 when constructing with
+  /// [CupertinoFormSection.centered].
+  final double borderRadius;
+
+  @override
+  State<StatefulWidget> createState() => _CupertinoFormSectionState();
+}
+
+class _CupertinoFormSectionState extends State<CupertinoFormSection> {
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    const Divider longDivider = Divider(
+      endIndent: 0,
+      indent: 0,
+      color: CupertinoColors.separator,
+      height: 0.0,
+    );
+
+    const Divider shortDivider = Divider(
+      endIndent: 0,
+      indent: 15.0,
+      color: CupertinoColors.separator,
+      height: 0.0,
+    );
+
+    final List<Widget> children =
+        (widget._type == _CupertinoFormSectionType.base)
+            ? <Widget>[longDivider]
+            : <Widget>[];
+
+    widget.children
+        .sublist(0, widget.children.length - 1)
+        .forEach((Widget element) {
+      children.add(element);
+      children.add(shortDivider);
+    });
+
+    children.add(widget.children.last);
+    if (widget._type == _CupertinoFormSectionType.base) {
+      children.add(longDivider);
+    }
+
+    return DecoratedBox(
+      decoration: const BoxDecoration(
+        color: CupertinoColors.secondarySystemBackground,
+      ),
+      child: Column(
+        children: <Widget>[
+          Padding(
+            padding: widget.titlePadding,
+            child: Row(
+              children: <Widget>[
+                Text(
+                  widget.title!.toUpperCase(),
+                  style: const TextStyle(
+                    fontSize: 13.5,
+                    color: CupertinoColors.secondaryLabel,
+                  ),
+                ),
+                const Spacer()
+              ],
+            ),
+          ),
+          Padding(
+            padding: widget.rowsPadding,
```

Comment:
I'd avoid having 2 padding values that people have to keep in sync (i.e. make titlePadding 2 horizontally, make this 10 horizontally, then realize the title is really to the left). If this is customizable, maybe it should just be a margin around the whole thing.

Reply:
Since this is a container, we should be a bit more deliberate in saying margin for the space outside and padding for the space between the rounded container and its inner children.